### PR TITLE
Refactor  StreamSource and CloudStreamSource to extend more general RandomAccessStream

### PR DESCRIPTION
**Pull request by EmmanuelMess**
* RandomAccessStream abstact class that encapsulates generalites between CloudStreamSource and StreamSource
* StreamSource and CloudStreamSource now extend RandomAccessStream
* CloudStreamServer uses CloudStreamSource
* Remove StreamSource default constructor
* Updated StreamSourceTest and CloudStreamSourceTest

### CODE REVIEW

File: app/src/androidTest/java/com/amaze/filemanager/asynchronous/services/ftp/FtpServiceEspressoTest.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ private void loginAndVerifyWith(FTPClient ftpClient) throws IOException {
         ftpClient.logout();
         ftpClient.disconnect();
 
-        assertTrue("Download folder not found on device. Either storage is not available, or something is really wrong with FtpService. Check logcat.",
+        assertTrue("Download folder not found on device. Either storage is not availableExact, or something is really wrong with FtpService. Check logcat.",
```
Comment (TranceLove) [inline app/src/androidTest/java/com/amaze/filemanager/asynchronous/services/ftp/FtpServiceEspressoTest.java]: availableExact -> available

File: app/src/main/java/com/amaze/filemanager/activities/MainActivity.java [hunk 2]
```diff
@@ -609,13 +609,13 @@ private void initializeInteractiveShell() {
     }
 
     /**
-     * Returns all available SD-Cards in the system (include emulated)
+     * Returns all availableExact SD-Cards in the system (include emulated)
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/activities/MainActivity.java]: availableExact -> available

File: app/src/main/java/com/amaze/filemanager/activities/MainActivity.java [hunk 3]
```diff
@@ -609,13 +609,13 @@ private void initializeInteractiveShell() {
...
      * <p>
      * Warning: Hack! Based on Android source code of version 4.3 (API 18)
      * Because there is no standard way to get it.
      * TODO: Test on future Android versions 4.4+
      *
-     * @return paths to all available SD-Cards in the system (include emulated)
+     * @return paths to all availableExact SD-Cards in the system (include emulated)
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/activities/MainActivity.java]: availableExact -> available

File: app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/ssh/PemToKeyPairTask.java [hunk 4]
```diff
@@ -134,7 +134,7 @@ public boolean shouldRetry(Resource<?> resource) {
                 }
             }
             if(retval == null)
-                throw new IOException("No converter available to parse selected PEM");
+                throw new IOException("No converter availableExact to parse selected PEM");
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/ssh/PemToKeyPairTask.java]: availableExact -> available

File: app/src/main/java/com/amaze/filemanager/filesystem/Operations.java [hunk 5]
```diff
@@ -54,13 +54,13 @@
 
         /**
          * Callback fired when creating new file/directory and required storage access framework permission
-         * to access SD Card is not available
+         * to access SD Card is not availableExact
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/filesystem/Operations.java]: availableExact -> available

File: app/src/main/java/com/amaze/filemanager/filesystem/Operations.java [hunk 6]
```diff
@@ -54,13 +54,13 @@
...
          */
         void launchSAF(HybridFile file);
 
         /**
          * Callback fired when renaming file and required storage access framework permission to access
-         * SD Card is not available
+         * SD Card is not availableExact
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/filesystem/Operations.java]: availableExact -> available

File: app/src/main/java/com/amaze/filemanager/filesystem/RootHelper.java [hunk 7]
```diff
@@ -259,7 +259,7 @@ private static boolean isDirectory(HybridFileParcelable path) {
      * Get a list of files using shell, supposing the path is not a SMB/OTG/Custom (*.apk/images)
      * TODO: Avoid parsing ls
      *
-     * @param root            whether root is available or not
+     * @param root            whether root is availableExact or not
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/filesystem/RootHelper.java]: availableExact -> available

File: app/src/main/java/com/amaze/filemanager/filesystem/RootHelper.java [hunk 8]
```diff
@@ -275,7 +275,7 @@ private static boolean isDirectory(HybridFileParcelable path) {
      * Get files using shell, supposing the path is not a SMB/OTG/Custom (*.apk/images)
      * TODO: Avoid parsing ls
      *
-     * @param root            whether root is available or not
+     * @param root            whether root is availableExact or not
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/filesystem/RootHelper.java]: availableExact -> available

File: app/src/main/java/com/amaze/filemanager/filesystem/ssh/Statvfs.java [hunk 9]
```diff
@@ -158,7 +158,7 @@ public long diskSize()
         }
 
         /**
-         * Return disk free space (filesystem block size * available filesystem blocks).
+         * Return disk free space (filesystem block size * availableExact filesystem blocks).
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/filesystem/ssh/Statvfs.java]: availableExact -> available

File: app/src/main/java/com/amaze/filemanager/fragments/preference_fragments/PrefFrag.java [hunk 10]
```diff
@@ -96,7 +96,7 @@ public void onCreate(Bundle savedInstanceState) {
 
         if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN_MR2 ||
                 sharedPref.getBoolean(PreferencesConstants.PREFERENCE_CRYPT_FINGERPRINT, false)) {
-            // encryption feature not available
+            // encryption feature not availableExact
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/fragments/preference_fragments/PrefFrag.java]: availableExact -> available

File: app/src/main/java/com/amaze/filemanager/ui/views/drawer/Drawer.java [hunk 11]
```diff
@@ -87,7 +87,7 @@
 
     private ActionViewStateManager actionViewStateManager;
     private boolean isSomethingSelected;
-    private volatile int phoneStorageCount = 0; // number of storage available (internal/external/otg etc)
+    private volatile int phoneStorageCount = 0; // number of storage availableExact (internal/external/otg etc)
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/ui/views/drawer/Drawer.java]: availableExact -> available

File: app/src/main/java/com/amaze/filemanager/utils/files/FileUtils.java [hunk 12]
```diff
@@ -143,7 +143,7 @@ public static long folderSize(SmbFile directory) {
      * Use recursive <code>ls</code> to get folder size.
      *
      * It is slow, it is stupid, and may be inaccurate (because of permission problems).
-     * Only for fallback use when <code>du</code> is not available.
+     * Only for fallback use when <code>du</code> is not availableExact.
```
Comment (TranceLove) [inline app/src/main/java/com/amaze/filemanager/utils/files/FileUtils.java]: availableExact -> available

File: app/src/main/res/layout/main_frag.xml [hunk 13]
```diff
@@ -27,7 +27,7 @@
 
     <!--
      As the main content view, the view below consumes the entire
-     space available using match_parent in both dimensions.
+     space availableExact using match_parent in both dimensions.
```
Comment (TranceLove) [inline app/src/main/res/layout/main_frag.xml]: availableExact -> available

**Comment by EmmanuelMess** [comment]
Ups, :smile:

**Comment by EmmanuelMess** [comment]
@TranceLove sorry for making you search for every excess replacement.

**Pull request merged**